src/sentry/static/sentry/app/components/suggestProjectCTA.tsx

```diff
@@ -50,29 +57,146 @@ class SuggestProjectCTA extends React.Component<Props> {
     if (!requestEntry) {
       return '';
     }
+    //find the user agent header out of our list of headers
     const userAgent = (requestEntry as EntryRequest)?.data?.headers?.find(
       item => item[0].toLowerCase() === 'user-agent'
     );
     if (!userAgent) {
       return '';
     }
+    //check if any of our mobile agent headers matches the event mobile agent
     return (
       MOBILE_USER_AGENTS.find(mobileAgent =>
         userAgent[1]?.toLowerCase().includes(mobileAgent.toLowerCase())
       ) ?? ''
     );
   }
+
   //check our projects to see if there is a mobile project
   get hasMobileProject() {
     return this.props.projects.some(project =>
       MOBILE_PLATFORMS.includes(project.platform || '')
     );
   }
 
+  get showCTA() {
+    const {promptIsLoaded, isDismissed} = this.state;
+
+    /**
+     * conditions to show prompt:
+     * 1. User agent matches mobile
+     * 2. No mobile project
+     * 3. CTA is not dimissed
+     * 4. We've loaded the data from the backend for the prompt
+     */
+    return (
+      !!this.matchedUserAgentString &&
+      !this.hasMobileProject &&
+      !isDismissed &&
+      promptIsLoaded
+    );
+  }
+
+  async fetchData() {
+    const {api, organization} = this.props;
+
+    //check our prompt backend
+    //no need to catch error since we have error boundary wrapping
+    const promptData = await promptsCheck(api, {
+      organizationId: organization.id,
+      feature: 'suggest_mobile_project',
+    });
+    const isDismissed = promptIsDismissed(promptData);
+
+    //set the new state
+    this.setState({
+      isDismissed,
+      promptIsLoaded: true,
+    });
+
+    const matchedUserAgentString = this.matchedUserAgentString;
+
+    //now record the results
+    trackAdvancedAnalyticsEvent(
```

Comment:
This is personal preference, but I kinda like to pull these analytic things up into their own functions outside of the component, so when you're reading the component, there's less visual noise in the code

Just my preference, not suggesting you need to do that, but maybe you agree

Reply:
@EvanPurkhiser Like a wrapper function around `trackAdvancedAnalyticsEvent`?

Reply:
hmm, actually I guess this is fine now with the way these work now, nvm